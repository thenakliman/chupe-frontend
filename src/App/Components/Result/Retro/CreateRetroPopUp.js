import React from 'react';
import propTypes from 'prop-types';
import {MINIMUM_VOTES} from './constants';

require('./Retros.css');

export class CreateRetroPopUp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {name: '', maximumVote: MINIMUM_VOTES};
    this.handleRetroName = this.handleRetroName.bind(this);
    this.handleMaximumVote = this.handleMaximumVote.bind(this);
    this.createRetro = this.createRetro.bind(this);
  }

  handleRetroName(name) {
    this.setState({name: name});
  }

  handleMaximumVote(maximumVote) {
    this.setState({maximumVote: maximumVote});
  }

  createRetro() {
    this.props.createRetro({
      name: this.state.name,
      maximumVote: this.state.maximumVote,
    });
    this.setState({name: '', maximumVote: MINIMUM_VOTES});
  }

  render() {
    return (
        <div className='retro-pop-up-container'>
          <div id='create-retro-container-id' className='create-retro-container'>
            <div className={'retro-button-container'}>
              <input className={'retro-creation-input'}
                     id={'create-retro-name'}
                     onChange={
                       (event) => this.handleRetroName(event.target.value)}
                     value={this.state.name}
              />
              <input className={'maximum-vote-count'}
                     id={'create-retro-maximum-vote-count'}
                     onChange={
                       (event) => this.handleMaximumVote(event.target.value)}
                     value={this.state.maximumVote}
              />
              <div className='create-retro-buttons-container'>
                <button id='create-retro-button-id'
                        className='create-retro-button'
                        type='button'
                        onClick={this.createRetro}
                >
                  Create Retro
                </button>
                <button id='cancel-create-retro-button-id'
                        className='create-retro-button'
                        onClick={this.props.closeCreatePopUp}
                        type='button'
                >
                  Cancel
                </button>
              </div>
            </div>
          </div>
        </div>);
  }
}

CreateRetroPopUp.propTypes = {
  retros: propTypes.array,
  closeCreatePopUp: propTypes.func.isRequired,
  createRetro: propTypes.func.isRequired,
};
