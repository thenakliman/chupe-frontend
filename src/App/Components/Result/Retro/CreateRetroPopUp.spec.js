/* eslint-disable */
import React from 'react';
import {CreateRetroPopUp} from './CreateRetroPopUp';
/* eslint-enable */
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Show create retro pop up component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<CreateRetroPopUp createRetro={jest.fn()}/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have empty name initially', () => {
    const wrapper = shallow(<CreateRetroPopUp createRetro={jest.fn()}/>);

    expect(wrapper.state().name).toEqual('');
  });

  it('should have maximum vote set to 1 initially', () => {
    const wrapper = shallow(<CreateRetroPopUp createRetro={jest.fn()}/>);

    expect(wrapper.state().maximumVote).toEqual(1);
  });

  it('should set name of retro in state', () => {
    const wrapper = shallow(<CreateRetroPopUp createRetro={jest.fn()}/>);
    const testName = 'some-name';
    wrapper.find('#create-retro-name').simulate(
        'change', {target: {value: testName}});

    expect(wrapper.state().maximumVote).toEqual(1);
  });

  it('should set maximum votes in state on change of input', () => {
    const wrapper = shallow(<CreateRetroPopUp createRetro={jest.fn()}/>);
    const maximumVote = 29;
    wrapper.find('#create-retro-maximum-vote-count').simulate(
        'change', {target: {value: maximumVote}});

    expect(wrapper.state().maximumVote).toEqual(maximumVote);
  });

  it('should call closeCreatePopUp on click of cancel', () => {
    const closeCreatePopUp = jest.fn();
    const wrapper = shallow(
        <CreateRetroPopUp createRetro={jest.fn()}
                          closeCreatePopUp={closeCreatePopUp}
        />);

    wrapper.find('#cancel-create-retro-button-id').simulate('click');
    expect(closeCreatePopUp).toHaveBeenCalledWith();
  });

  it('should call create retro', () => {
    const createRetro = jest.fn();
    const wrapper = shallow(<CreateRetroPopUp createRetro={createRetro}/>);
    const testName = 'some-name';
    wrapper.find('#create-retro-name').simulate(
        'change', {target: {value: testName}});

    const maximumVote = 29;
    wrapper.find('#create-retro-maximum-vote-count').simulate(
        'change', {target: {value: maximumVote}});

    wrapper.find('#create-retro-button-id').simulate('click');

    expect(createRetro).toHaveBeenCalledWith({
      name: testName,
      maximumVote: maximumVote,
    });
  });
});
