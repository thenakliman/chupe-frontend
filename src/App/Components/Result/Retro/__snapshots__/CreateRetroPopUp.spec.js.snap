// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show create retro pop up component should match snapshot 1`] = `
<div
  className="create-retro-container"
  id="create-retro-container-id"
>
  <div
    className="retro-button-container"
  >
    <input
      className="retro-creation-input"
      id="create-retro-name"
      onChange={[Function]}
      value=""
    />
    <input
      className="maximum-vote-count"
      id="create-retro-maximum-vote-count"
      onChange={[Function]}
      value={1}
    />
    <div
      className="create-retro-buttons-container"
    >
      <button
        className="create-retro-button"
        id="create-retro-button-id"
        onClick={[Function]}
        type="button"
      >
        Create Retro
      </button>
      <button
        className="create-retro-button"
        id="cancel-create-retro-button-id"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
